### ISSUE TITLE
Persona dialog displays partially in English

### ISSUE THREAD

**Issue by jds2501** [mozilla/persona#3499]
Migrated from https://bugzilla.mozilla.org/show_bug.cgi?id=880654.

Screenshot of Persona page on Spanish Firefox OS device

Description:
A user with their device language set to Spanish will see the Persona login dialog partially in English.

Repro Steps:

1) Select Settings, then Language, then change the language to Spanish.
2) Restart the device and click on http://translate.123done.org​
3) Observe the appearance of the Persona page.

Actual:
The Persona dialog displays partially in English.

Expected:
The Persona dialog displays in Spanish.

**Comment by inma610** [comment]
Thanks for migrating the issue, @jds2501. I didn't know browserid-related bugs were managed on github.

**Comment by callahad** [comment]
![personadialog](https://f.cloud.github.com/assets/24193/624390/89a359a0-cf84-11e2-8b90-8208c9134c57.png)

**Comment by ozten** [comment]
There is a [Spanish (es) locale in Browserid](https://localize.mozilla.org/es/browserid/).

I think this is a deployment issue with translate. We need to enable this language, run compress, and restart all node processes.

I don't have access to this box.

**Comment by jaredhirsch** [comment]
@ozten I'll add your pubkey to the box. Right now I'm going to try to enable the locale & run the steps you mentioned.

**Comment by jaredhirsch** [comment]
TL;DR - @jds2501 @inma610 can you try again? Should be fixed.

Details:

Hmm. Looking at [the l10n-all.json config file](https://github.com/mozilla/browserid/blob/train-2013.06.05/config/l10n-all.json) in the train, I see 'es' in the list of locales.

Onboard the translate server, I see that locale/es/LC_MESSAGES/client.po does contain the Spanish strings, for example:

``` shell
#: resources/static/dialog/views/pick_email.ejs:8
msgid "Sign in with Persona"
msgstr "Identificarse con Mozilla Persona"

#: resources/static/dialog/views/pick_email.ejs:11
msgid "Sign in as..."
msgstr "Identificarse como..."
```

`forever list` told me the processes had been running for seven hours--I didn't touch it overnight, presumably somebody added new strings and the server restarted after regenerating. Anyway, I restarted everything with `forever restartall`, set my firefox language to Spanish (es), and the dialog looks correct to me.

![screen shot 2013-06-07 at 10 47 09 am](https://f.cloud.github.com/assets/96396/625252/4e61834c-cf9a-11e2-9e3e-facece3e697f.png)

**Comment by ozten** [comment]
@6a68 thank you sir!

**Comment by jaredhirsch** [comment]
@ozten np! :beers:

@jds2501 @inma610 Closing. Feel free to reopen if it isn't fixed

**Comment by inma610** [comment]
Sorry guys, but on my Firefox OS the Persona dialog is still rendered in Spanglish.

**Comment by ozten** [comment]
Version number of FirefoxOS?

I'll get setup this afternoon to see what the `Accept-Language` header looks like from FirefoxOS.

/CC @jedp

**Comment by inma610** [comment]
Boot2Gecko 1.0.1.0. It's a Keon.

Thanks!

**Comment by jaredhirsch** [comment]
Hmm. I've got an unagi running Boot2Gecko 1.1.0.0-prerelease via OTA update, and I can't get the spinner to turn into a login button on translate.123done.org. I also don't get the popup when I click 'sign in' on translate.personatest.org. Am I missing something? @ozten you good to investigate this further?

**Comment by ozten** [comment]
I think this is the Geeks phone version of FirefoxOS, which means I can't test to debug.

@inma610 - I might need your help, I think about the best way to try to troubleshoot.

As a short term workaround, is using desktop Firefox to check translate good enough?

**Comment by ozten** [comment]
Using B2G 1.1 I see the Language Accept header looking good
![](http://dl.dropbox.com/u/10060532/Screenshots/L14U.png)

and 'es' appears partially localized.
![](http://dl.dropbox.com/u/10060532/Screenshots/7f8K.png)

![](http://dl.dropbox.com/u/10060532/Screenshots/OSkc.png)

With Geeksphone, can you go to http://bit.ly/115ditx and note what the Accept-Language header is, here. Thanks!

**Comment by ozten** [comment]
I didn't expect this... Here is Desktop firefox:
![](http://dl.dropbox.com/u/10060532/Screenshots/M8wD.png)

**Comment by ozten** [comment]
Okay, those strings, when used from the phone, come from firefoxos.personatest.org.

I'll update strings for all locales on that box.

**Comment by ozten** [comment]
Okay, fixed. Sorry about that.

**Comment by inma610** [comment]
Hi guys,

I just wanted to say THANK YOU, it's fixed! :)

**Comment by jaredhirsch** [comment]
\o/
